workspaces/src/result.rs

```diff
@@ -39,17 +41,29 @@ impl<T> CallExecution<T> {
     pub fn is_failure(&self) -> bool {
         self.details.is_failure()
     }
+
+    /// Checks whether the transaction was successful, if not returns a Result with the
+    /// corresponding error that occured.
+    pub fn ok(&self) -> Result<()> {
+        self.details.ok()
+    }
```

Comment:
Yeah this might not have been the best way of going about this one. I just added it because `CallExecution` was similar to `CallExecutionDetails`, but then again, we already have the `into_result` function here which does pretty much the same thing. I'll just get rid of this then